DataModel/Warning.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+
+namespace ChameleonSystem\UpgradeHelperBundle\DataModel;
+
+use Symfony\Component\Console\Formatter\OutputFormatterStyle;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Warning represents a potential illegal call to the container.
+ */
+final class Warning
+{
+    /** @var string The call is implicit using a variable */
+    public const TYPE_IMPLICIT = 'implicit';
+    /** @var string The call tries to fetch a non existing service */
+    public const TYPE_NON_EXISTENT = 'non-existent';
+
+    /**
+     * @var CallToContainer
+     */
+    public $call;
+    /**
+     * @var string One of self::TYPE_*
+     */
+    public $type;
+
+    /**
+     * format can be used to ouput the warning to the console
+     *
+     * @param OutputInterface $output
+     */
+    public function format(OutputInterface $output)
+    {
+        $outputStyle = new OutputFormatterStyle('magenta', 'default', ['bold']);
```

Comment:
If I recall correctly there is a formatter style class or method that adds default styles to the output - including at least `<warn>` and `<error>`

Reply:
I couldn't find one. But I may just be blind.

Reply:
:ok: